install/operator/pkg/syndesis/action/backup.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package action
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	cron "github.com/robfig/cron/v3"
```

Comment:
Would it make more sense to have a [k8s CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/), instead of the backup running within the operator process?

Reply:
no because I need to spin off a docker container for it

Reply:
@lgarciaaco can you elaborate that? You can spin the same image as the one used by the operator with perhaps a different entrypoint/arguments?

Reply:
Imho it is harder to work with pods than this way where the backup is tided up to the running binary. It is harder to develop and to debug. Why don't you tell me the benefits of doing it with a k8s cron job over using a cron withing the operator?

Reply:
I see backup and configuration as two separate concerns, for me it makes more sense to have them separate as any issues with the backup will be limited to the backup `CronJob`, likewise any issues with the operator would not jeopardize the backup. `CronJob` also gives the operators more flexibility than a feature we develop on our own, and matching existing features in our codebase unnecessary brings on more code to maintain.

Reply:
Those are arguable ... 
> as any issues with the backup will be limited to the backup CronJob, likewise any issues with the operator would not jeopardize the backup. CronJob

If the operator is jeopardize do I want to continue running backups? maybe syndesis is jeopardize all together? this seems to a bigger decision that requires some further discussion

>  also gives the operators more flexibility than a feature we develop on our own

Right now a feature I developed on my own allows me to do all I need. I don't need to use something different for the promise that in the future I might (or well might not) use some of it features. If I need those feature whatever it might be I can swap then, if the cron package I am using can't provide it.

Either way, my point is that using this library makes my development and test circle easier. I don't see any value using k8s cron over it, but if more engineers think it is better I will gladly change it

Reply:
> Those are arguable ...

I'm sorry, they're not, apart from the extend that you're willing to argue about them.

If the operator is failing because, say, it hits memory limits and gets OOMed, the backups are not run. Likewise if the backup ends up crashing the operator any pending configuration changes will not be applied.

Having the operator run the backups functionality that jeopardizes the backups and the rest of the operator functionality.

I'm okay with this getting merged to `master` and further improved upon, I'm not okay with this getting released in 1.9 as I think there will be issues in operating it.

Reply:
I am not willing to argue about anything. As I said, if other see a value right now or If we figure we actually need it, I will gladly change it.